aten/src/THNN/generic/SoftMarginCriterion.c

```diff
@@ -7,9 +7,19 @@ void THNN_(SoftMarginCriterion_updateOutput)(
   THTensor *input,
   THTensor *target,
   THTensor *output,
-  bool sizeAverage)
+  bool sizeAverage,
+  bool reduce)
 {
   THNN_CHECK_NELEMENT(input, target);
```

Comment:
Can you change this to `THNN_CHECK_SHAPE`? Checking the number of elements doesn't make that much sense today because we generally support broadcasting